CIScripts/Test/PesterHelpers/PesterHelpers.ps1

```diff
@@ -0,0 +1,67 @@
+﻿Import-Module Pester
+
+function Consistently {
+    <#
+    .SYNOPSIS
+    Utility wrapper for making sure that the assert is consistently true. It works by retrying
+    the assert every Interval seconds, up to Duration.
+    .PARAMETER ScriptBlock
+    ScriptBlock containing a Pester assertion.
+    .PARAMETER Interval
+    Interval (in seconds) between retries.
+    .PARAMETER Duration
+    Total time to spend between retries.
+    #>
+    Param (
+        [Parameter(Mandatory=$true)] [ScriptBlock] $ScriptBlock,
+        [Parameter(Mandatory=$false)] [int] $Interval = 1,
+        [Parameter(Mandatory=$false)] [int] $Duration = 3
+    )
+
+    if ($Duration -lt $Interval) {
+        throw "Duration must be longer than interval"
+    }
+    $NumRetries = $Duration / $Interval
+    foreach($i in 1..$NumRetries) {
+        & $ScriptBlock
+        Start-Sleep -s $Interval
+    }
+}
+
+function Eventually {
+    <#
+    .SYNOPSIS
+    Utility wrapper for making sure that the assert is eventually true. It works by retrying
+    the assert every Interval seconds, up to Duration. If until then, the assert is not true,
+    Eventually fails.
+    .PARAMETER ScriptBlock
+    ScriptBlock containing a Pester assertion.
+    .PARAMETER Interval
+    Interval (in seconds) between retries.
+    .PARAMETER Duration
+    Maximum to spend between retries. After this time, Eventually fails.
+    #>
+    Param (
+        [Parameter(Mandatory=$true)] [ScriptBlock] $ScriptBlock,
+        [Parameter(Mandatory=$false)] [int] $Interval = 1,
+        [Parameter(Mandatory=$false)] [int] $Duration = 3
```

Comment:
I think the default duration should be closer to inifinity (a minute or few). Or perhaps, make it mandatory?
Also, in my opinion, it'd be better to call this parameter `Timeout`.

Reply:
It should be as short as possible by default, so that the tests are as fast as possible. Parameter names are from Gomega.
Default values there are:
`[Eventually] The default value for the timeout is 1 second and the default value for the polling interval is 10 milliseconds`
`Consistently will monitor every 10 milliseconds for up to 100 milliseconds`

I decided not to make the defaults as low as those, since they are designed for single-program goroutine testing. 3 seconds seem fine for network stuff and services enabling/disabling.

Reply:
We've talked about whether `Duration` should be mandatory and we agree that it should, because if somebody has to use `Eventually`, they need to be doing it consciously and specify timeout explicitely.

Reply:
I'm worried this will lead to arbitrarily large timeouts. I'm thinking of adding a script analyzer rule to check for this in another PR... but ok